File: Go/41.FirstMissingPositive.go
func firstMissingPositive(nums []int) int {
    sort.Ints(nums)
    //fmt.Println(nums)
    mp := 1
    for _,n := range(nums) {
        if n > 0 {
            if n == mp {
                //fmt.Println("equal case", mp)
                mp++
            } else if n > mp {
                //fmt.Println("larger case", mp)
                break
            }
        }
    }
    
    return mp
}

File: Python3/41.FirstMissingPositive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        i = 0
        n = len(nums)
        while i<n:
            if nums[i] > 0 and nums[i] < n and nums[nums[i]-1] != nums[i]:
                nums[nums[i]-1], nums[i] = nums[i], nums[nums[i]-1]
            else:
                i += 1
        for i, n in enumerate(nums):
            if n != i+1:
                return i+1
        return len(nums)+1
